src/seedu/addressbook/data/person/Address.java

```diff
@@ -6,26 +6,29 @@
  * Represents a Person's address in the address book.
  * Guarantees: immutable; is valid as declared in {@link #isValidAddress(String)}
  */
-public class Address {
+public class Address extends Contact {
 
     public static final String EXAMPLE = "123, some street";
     public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
     public static final String ADDRESS_VALIDATION_REGEX = ".+";
 
-    public final String value;
-    private boolean isPrivate;
 
     /**
      * Validates given address.
      *
      * @throws IllegalValueException if given address string is invalid.
      */
-    public Address(String address, boolean isPrivate) throws IllegalValueException {
-        this.isPrivate = isPrivate;
+    public Address(String address, boolean isPrivate){
+        super(address, isPrivate);
+    }
+    
+    public createAddress(String address, boolean isPrivate) throws IllegalValueException {
```

Comment:
Where are you calling the method createAddress? It would have been to include the validation in the super class or do it in the Address constructor it self using a common isValid(regex, message) method. 

I don't think the use of the createAddress is correct.